### ISSUE TITLE
Clean code architecture

### ISSUE THREAD

**Issue by rooparsh** [ACM-VIT/gitpositive#55]
I feel that the project should have a proper architecture and filing system. It's becoming quite tedious and growing out of scope to manage resources.

**Comment by gaganmalvi** [comment]
Sure, will assign.

**Comment by DarthBenro008** [comment]
@rooparsh thanks for taking this up! Do kindly post your ideas to rearchitect the app here and run by the Idea before opening an Pull Request/Getting started to work on it. We might split the issues into smaller issues if required depending upon your proposal

**Comment by rooparsh** [comment]
Hi @DarthBenro008 , if you see currently the project is not properly structured, contributors are creating files in main package itself without any hierarchy and not following any coding guidelines. This in turn is causing confusion about what and where a particular piece of code is. 

My proposal is to properly package the modules functionality wise, keep common utils separate and adhere to strict coding guidelines.

I plan to achieve this in sub parts; in first PR, I have targeted proper packaging and modularity. Then I will focus on revamping coding guidelines and then finally removing deprecated stuff.

**Comment by DarthBenro008** [comment]
That's great to hear! But could you share what are these "strict coding guidelines" (which guideline to be exact?) and what kind of project structure are you willing to follow? Could you please draw a small architecture diagram on which "module" pertains to which functionality? This would be really great for future contributors to understand the codebase and have a proper architecture of the codebase.

TL;DR: Could you document the architecture pattern that you would be implementing?

**Comment by rooparsh** [comment]
Closing issue since Pr is merged here [Link](https://github.com/ACM-VIT/gitpositive/pull/62)
